### ISSUE TITLE
LibreOffice Online (Collabora) now working (fix inside)

### ISSUE THREAD

**Issue by dam09fr** [pydio/pydio-core#1293]
Hello, I spent a lot of time trying to make Collabora CODE docker image working without success with Pydio ... and Owncloud.

Recently I succeed building LibreOffice Online (https://cgit.freedesktop.org/libreoffice/core/) with manual compilation and it works with Owncloud but not with Pydio !

So I found some problems I want to share with you and how I solved: 

1) **Double '/' in wopi URL (loolwsd.log)**
```
wsd-04064-4141 0:06:57.634193 [ client_ws_000d ] INF  Loading [//wopi/files/1/Example.odt] for session [000d] and jail [4418].| DocumentBroker.cpp:198
wsd-04064-4141 0:06:57.634209 [ client_ws_000d ] DBG  Loading from URI: http://pydio.mydomain.fr//wopi/files/1/Example.odt?access_token=eyJ...KE8&permission=edit| DocumentBroker.cpp:214
```
In file "**core/src/plugins/editor.libreoffice/Driver.php**", at lines 121-122 : 
```
	$uri = '/wopi/files/' . $repositoryId . $nodePath;
	$escapedUri = '/wopi/files/' . $repositoryId . $escapedNodePath;
```
I changed "/wopi/files/" by "wopi/files" : 
```
	$uri = 'wopi/files/' . $repositoryId . $nodePath;
	$escapedUri = 'wopi/files/' . $repositoryId . $escapedNodePath;
```

2) **Missing UserCanWrite for LOOL CheckFileInfo :** 
After I fixed the previous problem, I was able to open my documents but there were readonly!
So I checked again loolwsd.log and I found this : 
```
wsd-19080-19128 0:00:52.733202 [ client_ws_0005 ] TRC  WOPI::CheckFileInfo header for URI [https://pydio.mydomain.fr/wopi/files/1/Example.odt?access_token=eyJ...yRc&permission=edit]:
```
I followed this URL and I got this answer : 
```
{"BaseFileName":"Example.odt","OwnerId":"dam09fr","Size":13179,"UserId":"dam09fr","Version":"1479800672","UserFriendlyName":"dam09fr"}
```
As document edit was ok on Owncloud I check the same on the Onwcloud side and the answer was different because it contains : << "UserCanWrite":true >>
So I changed the file "**core/src/core/src/pydio/Core/Http/Wopi/Middleware.php**" at line 104 and I add the missing "UserCanWrite" : 
```
	"UserCanWrite" => true,
	"UserFriendlyName" => $userId
```

--
Now I LOOL is working on my side.
I hope this can help you, if you need other elements, feel free to ask me.

**Comment by 7omate** [comment]
Concerning your double slash wopi, what was your SERVER URL in Pydio settings? Did it end with a "/"?
Great that you got to work! Thanks for sharing, too bad it seems so cumbersome.

**Comment by dam09fr** [comment]
![Pydio config](http://nsa37.casimages.com/img/2016/11/22/16112201593661999.jpg)
This is the default config after a fresh installation of Pydio.
The end slash is present by default so it seems it doesn't need to be added to the wopi URL.

**Comment by 7omate** [comment]
This is probably defined in your apache config. But yeah those wopi lines look like they need some fixing

**Comment by dam09fr** [comment]
I think so ;-)
What about the "UserCanWrite" attibute ?

I would like to know if possible on which version of LOOL / Collabora you carry out your developments and tests. Since it will appear that this version differs from the docker images provided and that do not seem to work for everyone.

**Comment by dam09fr** [comment]
Fix commit : https://github.com/pydio/pydio-core/commit/5ad4512663f61d56ea9e17b3ee6f093ad0096233

**Comment by cdujeu** [comment]
Fixed 7.0.2
